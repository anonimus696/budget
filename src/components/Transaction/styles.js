import styled from 'styled-components'

export const Wrapper = styled.div`
background: ${({ value }) => value > 0 ? '#6ffa8d' : '#ff8383'};
border: 1px solid #dbdbdb;
border-radius: 3px;
padding: 5px;
margin-bottom: 10px;
display: flex;
justify-content: space-between;
align-items: center;
`;

Wrapper.displayName = 'TransactionWrapper'

export const TransactionDate = styled.div`
flex: 1 1 30%;

`;
TransactionDate.displayName = 'TransactionDate'


export const Value = styled.div`
flex: 1 1 30%;


`;
TransactionDate.displayName = 'Value'

export const Comment = styled.div`
flex: 1 1 40%;


`;
TransactionDate.displayName = 'Comment'


export const Icon = styled.span`
img{
    max-width: 20px;
    margin-right: 15px;
}


`;
Icon.displayName = 'Icon'