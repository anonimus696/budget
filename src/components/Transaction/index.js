import PropTypes from 'prop-types'
import { useCallback, useContext } from 'react'
import { memo } from 'react'
import { AppContext } from '../../providers/context'

import star from '../../assets/img/star.svg'
import filledStar from '../../assets/img/fill.svg'

import { Wrapper, TransactionDate, Value, Comment, Icon } from './styles.js'

const Transaction = memo(({ transaction: { id, value, date, comment, isStared }, onDelete, onStarClick }) => {
    const { state } = useContext(AppContext);

    const deleteItem = useCallback(() => onDelete(id), [id])

    return (
        <Wrapper value={value}>
            <Icon onClick={() => onStarClick(id)}>
                <img src={isStared ? filledStar : star} />
            </Icon>
            <TransactionDate>{date}</TransactionDate>
            <Value>{value.toFixed(2)}, {state.currency}</Value>
            <Comment>{comment}</Comment>
            <button onClick={deleteItem}>Delete</button>
        </Wrapper>
    );
})



Transaction.defaultProps = {
    transaction: {
        label: '',
        value: 0
    }
}


Transaction.propTypes = {
    transaction: PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.number
    })
}


export default Transaction;