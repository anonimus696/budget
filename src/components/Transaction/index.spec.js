import React from "react";
import { render } from "@testing-library/react";
import Transaction from ".";

describe('Transaction comment', () => {
    it('should show transaction', () => {
        const { asFragment } = render(<Transaction />);
        expect(asFragment()).toMatchSnapshot();
    })
})