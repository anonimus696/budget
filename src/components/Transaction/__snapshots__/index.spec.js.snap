// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction comment should show transaction 1`] = `
<DocumentFragment>
  <div
    class="sc-bcPKhP eBdqPb"
    value="0"
  >
    <div
      class="sc-grXZZQ dZjqqV"
    />
    <div
      class="sc-dkjKgF bOEvHs"
    >
      0.00
    </div>
    <div
      class="sc-hLclGa jBJIOt"
    />
  </div>
</DocumentFragment>
`;
